### PR TITLE
Add transforms to logged config

### PR DESCRIPTION
**Pull request by b-chu**
Uses the transformed version of the training config as the logged config. This is because we infer values like device_train_batch_size during transforms. The logged config is later used in callbacks like curriculum learning which should have access to all the [data in TrainConfig](https://github.com/mosaicml/llm-foundry/blob/38dcf1e6e25a05dbd4275402f168bac299b45d6d/llmfoundry/utils/config_utils.py#L95).

Manual run to test that logged_cfg matches the train_cfg values from transforms and curriculum learning callback works.

### CODE REVIEW

**Review by snarayan21 — COMMENTED** [review]
ooh nice. can you include the run name for the manual test (and highlight the differences in the logged configs, if possible)?

thanks!!

**Comment by b-chu** [comment]
logged_cfg before
dict_keys(['variables', 'max_seq_len', 'run_name', 'model', 'tokenizer', 'train_loader', 'scheduler', 'optimizer', 'algorithms', 'max_duration', 'eval_interval', 'eval_first', 'eval_subset_num_batches', 'global_train_batch_size', 'seed', 'device_eval_batch_size', 'device_train_microbatch_size', 'precision', 'fsdp_config', 'progress_bar', 'log_to_console', 'console_log_interval', 'callbacks', 'save_interval', 'merge'])

logged_cfg after
dict_keys(['variables', 'max_seq_len', 'run_name', 'model', 'tokenizer', 'train_loader', 'scheduler', 'optimizer', 'algorithms', 'max_duration', 'eval_interval', 'eval_first', 'eval_subset_num_batches', 'global_train_batch_size', 'seed', 'device_eval_batch_size', 'device_train_microbatch_size', 'precision', 'fsdp_config', 'progress_bar', 'log_to_console', 'console_log_interval', 'callbacks', 'save_interval', 'n_gpus', 'device_train_batch_size', 'device_train_grad_accum', 'merge'])

**Review by snarayan21 — APPROVED** [review]
lgtm

**Comment by b-chu** [comment]
Failing from transient HF timeouts :( @snarayan21 PR also needs to update branch

**Review by dakinggg — APPROVED** [review]
Ah sorry, forgot the callbacks use the logged config. Thanks!
